Foundation/NSJSONSerialization.swift

```diff
@@ -716,21 +716,16 @@ private struct JSONReader {
                 defer { intEndPointer.deallocate(capacity: 1) }
                 let doubleEndPointer = UnsafeMutablePointer<UnsafeMutablePointer<Int8>?>.allocate(capacity: 1)
                 defer { doubleEndPointer.deallocate(capacity: 1) }
-                
                 let intResult = strtol(startPointer, intEndPointer, 10)
                 let intDistance = startPointer.distance(to: intEndPointer[0]!)
                 let doubleResult = strtod(startPointer, doubleEndPointer)
                 let doubleDistance = startPointer.distance(to: doubleEndPointer[0]!)
-                
+                let roundedDouble = doubleResult.rounded()
                 guard intDistance > 0 || doubleDistance > 0 else {
                     return nil
                 }
-                
-                if intDistance == doubleDistance {
```

Comment:
Removing these lines seems suspect. First, `intResult` is now unused. Second--more importantly--there are values of `Int` that cannot be exactly represented in `Double` (those above 2e53). Was there a reason why you removed these lines?

You will almost certainly want the second `guard` statement below before proceeding to use `doubleResult`.